core/objects.go

```diff
@@ -493,6 +493,12 @@ type SuggestedWindow struct {
 	End   time.Time `json:"end"`
 }
 
+// IsWithin returns true if the given time is within the suggested window,
+// inclusive of the start and end times.
```

Comment:
IIRC there was discussion during the last check-in to make this a half open check e.g. inclusive of the start and exclusive of the end time.